pkg/trace/sampler/remote_rates.go

```diff
@@ -0,0 +1,225 @@
+package sampler
+
+import (
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config/remote/service"
+	"github.com/DataDog/datadog-agent/pkg/proto/pbgo"
+	"github.com/DataDog/datadog-agent/pkg/trace/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/metrics"
+	"github.com/DataDog/datadog-agent/pkg/trace/pb"
+	"github.com/DataDog/datadog-agent/pkg/trace/watchdog"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// RemoteRates sharded by signature. Allowing an independent feedback
+// loop per signature. RemoteRates adjusts sampling rates communicated
+// to the agent based on the observed traffic. It targets a TPS configured
+// remotely.
+type RemoteRates struct {
+	samplers   map[Signature]*Sampler
+	tpsTargets map[Signature]float64
+	mu         sync.RWMutex // protects concurrent access to samplers and tpsTargets
+
+	exit    chan struct{}
+	stopped chan struct{}
+}
+
+func newRemoteRates(conf *config.AgentConfig) *RemoteRates {
+	if !conf.RemoteRates {
+		return nil
+	}
+
+	remoteRates := &RemoteRates{
+		samplers: make(map[Signature]*Sampler),
+
+		exit:    make(chan struct{}),
+		stopped: make(chan struct{}),
+	}
+
+	err := service.NewGRPCSubscriber(
```

Comment:
```suggestion
	}
	remoteRates := &RemoteRates{
		samplers: make(map[Signature]*Sampler),
		exit:    make(chan struct{}),
		stopped: make(chan struct{}),
	}
	err := service.NewGRPCSubscriber(
```
I think there is a bit too much spacing here. WDYT?